nuxeo-admin-console-web/angular-app/src/app/features/elastic-search-reindex/services/elastic-search-reindex.service.ts

```diff
@@ -1,42 +1,60 @@
-import { NuxeoJSClientService } from "./../../../shared/services/nuxeo-js-client.service";
 import { Injectable } from "@angular/core";
 import { HttpClient } from "@angular/common/http";
 import { BehaviorSubject, Observable } from "rxjs";
 import { ReindexInfo } from "../elastic-search-reindex.interface";
+import { NuxeoJSClientService } from "../../../shared/services/nuxeo-js-client.service";
+import { finalize } from "rxjs/operators";
 
 @Injectable({
   providedIn: "root",
 })
 export class ElasticSearchReindexService {
   private elaticSearchReindexEndpoint = "management/elasticsearch/reindex";
   pageTitle: BehaviorSubject<string> = new BehaviorSubject("");
+  spinnerStatus: BehaviorSubject<boolean> = new BehaviorSubject(false);
+
   constructor(
     private http: HttpClient,
     private nuxeoJsClientService: NuxeoJSClientService
   ) {}
 
+  setSpinnerStatus(status: boolean): void {
+    this.spinnerStatus.next(status);
+  }
+
   performDocumentReindex(requestQuery: string | null): Observable<ReindexInfo> {
-    return this.http.post<ReindexInfo>(
-      `${this.nuxeoJsClientService.getApiUrl()}${
-        this.elaticSearchReindexEndpoint
-      }?query=${requestQuery}`,
-      {}
-    );
+    this.setSpinnerStatus(true);
```

Comment:
this should move to the point where this action is dispatched, in the component.